src/pointer-analysis/value_set_dereference.cpp

```diff
@@ -222,6 +224,19 @@ bool value_set_dereferencet::dereference_type_compare(
   {
     return true;
   }
+  else if(dereference_unwrapped->id() == ID_pointer &&
+          object_unwrapped->id() != ID_pointer)
+  {
+    ui_message_handlert mh;
+    messaget message(mh);
+    message.debug() << "value_set_dereference: the types have different "
+                       "level of ID_pointer levels"
+                    << messaget::endl;
+    message.debug() << " object_type: " << object_type.pretty()
+                    << messaget::endl;
+    message.debug() << " dereference_type: " << dereference_type.pretty()
+                    << messaget::eom;
+  }
```

Comment:
I was unsure whether this 'forced' debug is a good idea actually. The thing is that it took us a long time to find the problem in symex so for future cases this output could save us time. But I agree it's not ideal.

Reply:
"forced debug" is better known as "warning" :-)  So if you want to keep it in (and you have given good reasons for doing so), then please make sure there is a message handler in place so that proper output can be generated. If that's a bigger project than is feasible right now, you might just create a fresh PR with just this commit in it?

Reply:
I meant forced in that I create a message handler especially for this :)

Reply:
I'm not sure I understand; depending on who should be able to control this output it would either be guarded by `#ifdef DEBUG` or be passed a configured message handler.

Reply:
I wanted to do it via a message handler (and changing the `debug()` above into `warning()`) but passing the message handler is a nightmare... so I did it as `#ifdef DEBUG` instead.